src/explorer/WebAppTreeItem.ts

```diff
@@ -93,66 +94,59 @@ export class WebAppTreeItem extends SiteTreeItem {
         const siteConfig = taskResults[2];
-        const script = scriptTemplate.replace('%SUBSCRIPTION_NAME%', subscription.displayName)
-            .replace('%RG_NAME%', rg.name)
-            .replace('%LOCATION%', rg.location)
-            .replace('%PLAN_NAME%', plan.name)
-            .replace('%PLAN_SKU%', plan.sku.name)
-            .replace('%SITE_NAME%', site.name)
-            .replace('%RUNTIME%', siteConfig.linuxFxVersion);
-        await new Promise<void>((resolve, reject) => {
-            fs.writeFile(uri.fsPath, script, err => {
-                if (err) {
-                    reject(err);
-                } else {
-                    resolve();
-                }
-            });
-        });
+        const appSettings = taskResults[3];
+
+        let script: string;
+
+        if (!siteConfig.linuxFxVersion) {
+            const scriptTemplate = await this.loadScriptTemplate('windows-default.sh');
+            script = scriptTemplate.replace('%SUBSCRIPTION_NAME%', subscription.displayName)
+                .replace('%RG_NAME%', rg.name)
+                .replace('%LOCATION%', rg.location)
+                .replace('%PLAN_NAME%', plan.name)
+                .replace('%PLAN_SKU%', plan.sku.name)
+                .replace('%SITE_NAME%', site.name);
+        } else if (siteConfig.linuxFxVersion.toLowerCase().startsWith('linux')) {
+            const scriptTemplate = await this.loadScriptTemplate('linux-default.sh');
+            script = scriptTemplate.replace('%SUBSCRIPTION_NAME%', subscription.displayName)
+                .replace('%RG_NAME%', rg.name)
+                .replace('%LOCATION%', rg.location)
+                .replace('%PLAN_NAME%', plan.name)
+                .replace('%PLAN_SKU%', plan.sku.name)
+                .replace('%SITE_NAME%', site.name)
+                .replace('%RUNTIME%', siteConfig.linuxFxVersion);
+        } else if (siteConfig.linuxFxVersion.toLowerCase().startsWith('docker')) {
+            const scriptTemplate = await this.loadScriptTemplate('docker-image.sh');
+            const serverUrl = appSettings.properties['DOCKER_REGISTRY_SERVER_URL'];
+            const serverUser = appSettings.properties['DOCKER_REGISTRY_SERVER_USERNAME'];
+            const serverPwd = appSettings.properties['DOCKER_REGISTRY_SERVER_PASSWORD'];
+            const containerParameters =
+                (serverUrl ? `SERVERURL="${serverUrl}"\n` : '') +
+                (serverUser ? `SERVERUSER="${serverUser}"\n` : '') +
+                (serverPwd ? `SERVERPASSWORD="*****"\n` : '')
+            const containerCmdParameters =
+                (serverUrl ? '--docker-registry-server-url $SERVERURL ' : '') +
+                (serverUser ? '--docker-registry-server-user $SERVERUSER ' : '') +
+                (serverPwd ? '--docker-registry-server-password $SERVERPASSWORD ' : '')
+            script = scriptTemplate.replace('%SUBSCRIPTION_NAME%', subscription.displayName)
+                .replace('%RG_NAME%', rg.name)
```

Comment:
nit: you could move the common 'replace's after the if/else:
```
script = script.replace('%SUBSCRIPTION_NAME%', subscription.displayName)
                .replace('%RG_NAME%', rg.name)
                .replace('%LOCATION%', rg.location)
                .replace('%PLAN_NAME%', plan.name)
                .replace('%PLAN_SKU%', plan.sku.name)
                .replace('%SITE_NAME%', site.name)
```
I think those are all common ^